src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -619,12 +621,22 @@ namespace smt::noodler {
         conjuncts.push_back(conv_form_with_precision.first);
         precision = conv_form_with_precision.second;
 
+        bool has_lia_reducible_predicates = (!solution.length_sensitive_vars.empty() || !this->not_contains.get_predicates().empty() || !this->disequations.get_predicates().empty());
+        if (!has_lia_reducible_predicates) {
+            return {LenNode(LenFormulaType::AND, conjuncts), precision};
+        }
 
         // get the LIA formula describing solutions for special predicates
         conjuncts.push_back(get_formula_for_ca_diseqs());
         auto not_cont_prec = get_formula_for_not_contains();
+
         if (not_cont_prec.second == LenNodePrecision::PRECISE || not_cont_prec.second == precision) {
+            // Formula for not-contains is at least as precise as is the current precision
+            conjuncts.push_back(not_cont_prec.first);
+        } else if (this->disequations.get_predicates().empty()) {
+            // We do not have any disequations at all, so we can just overwrite the precision with whatever came out of not-contains
             conjuncts.push_back(not_cont_prec.first);
+            precision = not_cont_prec.second;
```

Comment:
This seems incorrect, precision from conversions could be thrown away,

Reply:
Yes, you are correct. I have added a library function `get_resulting_precision_for_conjunction` that implements the precision-merging logic when taking conjunctions.

Reply:
Hmm, that's a good idea, but the logic in the function seems incorrect, I'll add a new comment.